samples/remote_attestation/common/dispatcher.cpp

```diff
@@ -182,10 +182,173 @@ int ecall_dispatcher::verify_report_and_set_pubkey(
     return ret;
 }
 
+int ecall_dispatcher::establish_secure_channel(uint8_t** key, size_t* key_size)
+{
+    uint8_t encrypted_key_buf[512];
+    uint8_t signature[256];
+    uint8_t digest[32];
+    size_t encrypted_key_size = 256;
+    size_t signature_size = 256;
+    size_t total_size = 512;
+
+    int ret = 1;
+
+    if (m_initialized == false)
+    {
+        TRACE_ENCLAVE("ecall_dispatcher initialization failed.");
+        goto exit;
+    }
+
+    // Step 1 - Create an ephemeral symmetric key
+    if (oe_random(m_enclave_config->sym_key, 32) != OE_OK)
+    {
+        TRACE_ENCLAVE("ecall_dispatcher initialization failed to generate "
+                      "ephemeral symmetric key.");
+        goto exit;
+    }
+
+    TRACE_ENCLAVE("enclave: establish_secure_channel: Generated random "
+                  "symmetric key successfully");
+
+    // Step 2- Encrypt the symmetric key with the other enclave's public key
+    // TO DO - PREFIX with counter/seq number to prevent replay attacks
```

Comment:
If you want to do it this way, this is not optional and needs to be part of this changeset.

Reply:
Yep, agree and plan to implement.

Reply:
Implemented